sql/range.go

```diff
@@ -21,15 +21,35 @@ import (
 )
 
 // RangeCollection is a collection of ranges that represent different (non-overlapping) filter expressions.
-type RangeCollection []Range
+type RangeCollection interface {
+	Equals(otherCollection RangeCollection) (bool, error)
+	DebugString() string
```

Comment:
Should embed the appropriate interfaces for String and DebugString instead of the methods directly